src/main/scala/org/scalajs/tools/tsimporter/Trees.scala

```diff
@@ -107,7 +107,7 @@ object Trees {
 
   case class BooleanLiteral(value: Boolean) extends Literal
 
-  case class NumberLiteral(value: Double) extends Literal
+  case class NumberLiteral(value: Double, isValidInt: Boolean) extends Literal
```

Comment:
This structure is a bit weird, IMO. Have you considered having two separate
```scala
case class IntLiteral(value: Int) extends Literal
case class DoubleLiteral(value: Double) extends Literal
```
instead?

Reply:
Refactored in https://github.com/sjrd/scala-js-ts-importer/pull/47/commits/8bf944f1ce9b26122978dad991a3ac5891f43e66